src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -750,5 +733,328 @@ private static string ToHtmlCase(string name)
+
+                    valueStart = _index;
+                    while (!At(quote))
+                    {
+                        if (AtEnd)
+                        {
+                            errorSink.OnError(SourceLocation.Zero, "TODO: Mismatching quotes", 0);
+                            return null;
+                        }
+
+                        Next();
+                    }
+                    valueEnd = _index;
+
+                    // Move past the end quote;
+                    Next();
+                }
+                else
+                {
+                    valueStart = _index;
+                    while (!AtEnd && !char.IsWhiteSpace(Current) && !At(']'))
+                    {
+                        Next();
+                    }
+                    valueEnd = _index;
+                }
+
+                PassOptionalWhitespace();
+
+                var value = _requiredAttributes.Substring(valueStart, valueEnd - valueStart);
+
+                return value;
+            }
+
+            private TagHelperRequiredAttributeDescriptor ParseCSSSelector(ErrorSink errorSink)
+            {
+                Debug.Assert(Current == '[');
+
+                // Move past '['.
+                Next();
+                PassOptionalWhitespace();
+
+                var attributeName = ParseCSSAttributeName(errorSink);
+
+                if (attributeName == null)
+                {
+                    // Couldn't parse attribute name.
+                    return null;
+                }
+
+                if (!ValidateName(attributeName, targetingAttributes: true, errorSink: errorSink))
+                {
+                    return null;
+                }
+
+                if (!EnsureNotAtEnd(errorSink, ']'))
+                {
+                    return null;
+                }
+
+                var valueOperator = default(char);
+                if (At('='))
+                {
+                    valueOperator = ParseCSSValueOperator(errorSink);
+                }
+
+                if (!EnsureNotAtEnd(errorSink, ']'))
+                {
+                    return null;
+                }
+
+                var value = ParseCSSValue(errorSink);
+
+                if (value == null)
+                {
+                    // Couldn't parse value
+                    return null;
+                }
+
+                if (At(']'))
+                {
+                    // Move past the ending bracket.
+                    Next();
+                }
+                else
+                {
+                    errorSink.OnError(SourceLocation.Zero, "TODO: Could not find matching ']' for required attribute.", 0);
+                    return null;
+                }
+
+                return new TagHelperRequiredAttributeDescriptor
+                {
+                    Name = attributeName,
+                    Value = value,
+                    Operator = valueOperator,
+                    IsCSSSelector = true,
```

Comment:
I'll rename, going to leave as `true` in the `[name]` case since it behaves differently than simply having `name`, aka `[name-*]` is not the same as `name-*`.